Sources/SwiftSyntax/Utils.swift

```diff
@@ -102,3 +102,29 @@ extension RawUnexpectedNodesSyntax {
     self.init(raw: raw)
   }
 }
+
+extension TypeSyntax {
+  public var isVoid: Bool {
+    switch self.as(TypeSyntaxEnum.self) {
+    case .identifierType(let identifierType) where identifierType.name.text == "Void": return true
+    case .tupleType(let tupleType) where tupleType.elements.isEmpty: return true
+    case .memberType(let memberType) where memberType.name.text == "Void": return memberType.baseType.isSwiftCoreModule
+    default: return false
+    }
+  }
+
+  public var isSwiftInt: Bool {
+    switch self.as(TypeSyntaxEnum.self) {
+    case .identifierType(let identifierType) where identifierType.name.text == "Int": return true
+    case .memberType(let memberType) where memberType.name.text == "Int": return memberType.baseType.isSwiftCoreModule
+    default: return false
+    }
+  }
```

Comment:
This is largely a copy of the `isVoid` implementation above. I'm concerned that this approach to the API doesn't really scale: are we going to end up growing `isSwiftUInt`, `isSwiftBool`, `isSwiftSet`, and others as separate properties? That doesn't seem like a desirable end state.

Reply:
I found a way to generalise it to types where the canonical name only contains `MemberTypeSyntax` or `IdentifierTypeSyntax`. This doesn't include `Void` since it's a tuple, and there doesn't seem to be a good way to generalise meta types for tuples; I couldn't find anything like `Tuple.Type`, so handling tuple types would involve enumerating every arity. Not sure how to handle type shadowing either, since there doesn't seem to be a way to reify `module<Swift>.self`.